File: Src/AuthServer/app/src/components/features/business/apps/models/app-vm.ts
import {ExternalProvider} from "../../../auth/external-log-in/models/external-provider";
import {LocalAccountSettings} from "./local-account-settings";


export class AppVm {

    public id: string;
    public name: string;
    public key: string;
    public isLocalAccountEnabled: boolean;
    public isRememberLogInEnabled: boolean;
    public isSecurityQuestionsEnabled: boolean;
    public emailSettings: LocalAccountSettings;
    public phoneSettings: LocalAccountSettings;
    public externalProviders: ExternalProvider[] = [];

}

File: Src/AuthServer/Services/Apps/AppVm.cs
using System;
using System.Collections.Generic;

namespace AuthGuard.Services.Apps
{
    public class AppVm
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Key { get; set; }
        public bool IsLocalAccountEnabled { get; set; }
        public bool IsRememberLogInEnabled { get; set; }
        public bool IsSecurityQuestionsEnabled { get; set; }
        public LocalAccountSettingsVm EmailSettings { get; set; }
        public LocalAccountSettingsVm PhoneSettings { get; set; }
        public IEnumerable<ExternalProviderVm> ExternalProviders { get; set; }
    }
}